### PR TITLE
#1928 new test order

### PR DESCRIPTION
**Pull request by jorgegorka**
https://find.plan.io/issues/1928

### CODE REVIEW

File: app/assets/javascripts/components/encounter_form.js.jsx [hunk 1]
```diff
@@ -11,16 +11,17 @@ var BaseEncounterForm = {
...
-	      window.location.href = '/encounters/' + this.state.encounter.id;
-	    }
+        window.location.href = '/encounters/' + this.state.encounter.id;
+      }
     };
 
     if (this.state.encounter.id) {
       this._ajax('PUT', '/encounters/' + this.state.encounter.id, callback);
     } else {
+      console.log('creating encounter');
```
Comment (tripledub) [inline app/assets/javascripts/components/encounter_form.js.jsx]: Do you want this left in here?

Comment (jorgegorka) [inline app/assets/javascripts/components/encounter_form.js.jsx]: Ouch

File: app/assets/javascripts/components/fresh_tests_encounter_form.js.jsx [hunk 2]
```diff
@@ -5,6 +5,23 @@ var FreshTestsEncounterForm = React.createClass(_.merge({
...
+    var foo = '';
+
+    if (clk === 0) { foo = 'diag'; }
+
+    if (clk === 1) { foo = 'follow'; }
+
+    this.setState(React.addons.update(this.state, {
+      encounter: {
+        exam_reason: {
+          $set: foo
```
Comment (tripledub) [inline app/assets/javascripts/components/fresh_tests_encounter_form.js.jsx]: `foo`? something more meaningful perhaps?

Comment (jorgegorka) [inline app/assets/javascripts/components/fresh_tests_encounter_form.js.jsx]: That's the original variable name. I didn't create it, but I'll change it anyway.

File: app/controllers/encounters_controller.rb [hunk 3]
```diff
@@ -166,7 +160,7 @@ def determine_referal
       referer_check = referer =~ /\/patients\/\d/
        if  (referer_check != nil) || (params['test_order_page_mode'] == 'cancel')
         @show_edit_encounter=false
-        @show_cancel_encounter=true if (@encounter != nil) && (Encounter.statuses[@encounter.status] != Encounter.statuses["inprogress"])
+        @show_cancel_encounter=true if @encounter && (Encounter.statuses[@encounter.status] != Encounter.statuses["inprogress"])
```
Comment (tripledub) [inline app/controllers/encounters_controller.rb]: If we are correcting bad style here then `@show_cancel_encounter = true` also looks better (nit picking sorry)

**Pull request merged**